### PR TITLE
add max_threads_per_core setting

### PR DESCRIPTION
**Pull request by xufang-lisa**
### Details:
 - *add max_threads_per_core setting*

### Tickets:
 - *117357*

### CODE REVIEW

File: src/inference/tests/unit/cpu_stream_info_test.cpp [hunk 1]
```diff
@@ -361,7 +372,31 @@ LinuxCpuStreamTypeCase _1sockets_4cores_binding = {
         MAIN_CORE_PROC,
     },
     {0, 0, 0, 0},
+    {1, 1, 1, 1},
 };
+
+LinuxCpuStreamTypeCase _1sockets_4cores_hyperthreading = {
```
Comment (sunxiaoxia2022) [inline src/inference/tests/unit/cpu_stream_info_test.cpp:378]: I think this test case should correspond to _1sockets_4cores_nobinding. The parameter _core_type should changed to ALL_PROC.

Comment (xufang-lisa) [reply src/inference/tests/unit/cpu_stream_info_test.cpp]: thanks, updated

**Review by wangleis — APPROVED** [review]

**Comment by wangleis** [comment]
Based on https://github.com/openvinotoolkit/openvino/pull/18863

**Review by sunxiaoxia2022 — APPROVED** [review]

File: src/inference/src/dev/threading/cpu_streams_executor.cpp [hunk 2]
```diff
@@ -132,21 +132,29 @@ struct CPUStreamsExecutor::Impl {
...
+                                   const int max_threads_per_core) {
             _numaNodeId = (_impl->_usedNumaNodes.size() == 1 && _impl->_usedNumaNodes.at(0) == -1)
                               ? -1  // macOS
                               : std::max(0, numa_node_id);
             _socketId = get_socket_by_numa_node(_numaNodeId);
             if (stream_type == STREAM_WITHOUT_PARAM) {
-                _taskArena.reset(new custom::task_arena{concurrency});
+                _taskArena.reset(new custom::task_arena{custom::task_arena::constraints{}
+                                                            .set_max_concurrency(concurrency)
+                                                            .set_max_threads_per_core(max_threads_per_core)});
```
Comment (dmitry-gorokhov) [inline src/inference/src/dev/threading/cpu_streams_executor.cpp:144]: As we discussed lets clarify `set_max_threads_per_core` behavior with legacy TBB before merge.

Comment (xufang-lisa) [reply src/inference/src/dev/threading/cpu_streams_executor.cpp:144]: For legacy TBB, openvino uses static tbbbind library and set max_threads_per_core to tbbbind directly.

Comment (wangleis) [reply src/inference/src/dev/threading/cpu_streams_executor.cpp:144]: @dmitry-gorokhov Please find the detail at CVS-117357

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**